File: app/src/main/java/com/taipei/yanghaobo/kunu/data/DataConverters.java
package com.taipei.yanghaobo.kunu.data;

import android.arch.persistence.room.TypeConverter;
import android.support.annotation.Nullable;

import java.util.Date;


public class DataConverters {

    @Nullable
    @TypeConverter
    public static Date toDate(@Nullable Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @Nullable
    @TypeConverter
    public static Long toTimestamp(@Nullable Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/com/taipei/yanghaobo/kunu/data/DataConverters.kt
package com.taipei.yanghaobo.kunu.data

import androidx.room.TypeConverter

import java.util.Date

object DataConverters {

  @TypeConverter
  fun toDate(timestamp: Long?): Date? {
    return if (timestamp == null) null else Date(timestamp)
  }

  @TypeConverter
  fun toTimestamp(date: Date?): Long? {
    return date?.time
  }
}
